[fontconfig] build failure

**Host Environment**
- OS: Linux Manjaro
- Compiler: g++

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install fontconfig`

**Failure logs** 
```--- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Generating configure for x64-linux
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:108 (message):
    Command failed: /usr/bin/autoreconf -vfi
    Working Directory: /home/venny/projects/vctest/vcpkg/buildtrees/fontconfig/src/75c13b2421-4651fd65ce.clean/
    Error code: 1
    See logs for more information:
      /home/venny/projects/vctest/vcpkg/buildtrees/fontconfig/autoconf-x64-linux-out.log
      /home/venny/projects/vctest/vcpkg/buildtrees/fontconfig/autoconf-x64-linux-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:536 (vcpkg_execute_required_process)
  ports/fontconfig/portfile.cmake:25 (vcpkg_configure_make)
  scripts/ports.cmake:136 (include)


Error: Building package fontconfig:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: fontconfig:x64-linux
  Vcpkg version: 2020.11.12-unknownhash
```

The mentioned log files are attached.
[autoconf-x64-linux-err.log](https://github.com/microsoft/vcpkg/files/5779601/autoconf-x64-linux-err.log)
[autoconf-x64-linux-out.log](https://github.com/microsoft/vcpkg/files/5779602/autoconf-x64-linux-out.log)

**Additional context**
I tried installing the nana package, but it failed at fontconfig.
Installing fontconfig directly renders the same result.

Thanks for the quick reply. With `automake` I managed to complete the install but I haven't been able to build with this package.
This is the output from make:
```
Scanning dependencies of target calc
[ 50%] Building CXX object CMakeFiles/calc.dir/calc.cpp.o
[100%] Linking CXX executable calc
/usr/bin/ld: ../../vcpkg/installed/x64-linux/lib/libfontconfig.a(fccache.o): undefined reference to symbol 'uuid_generate_random@@UUID_1.0'
/usr/bin/ld: /usr/lib/libuuid.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/calc.dir/build.make:127: calc] Error 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/calc.dir/all] Error 2
make: *** [Makefile:103: all] Error 2
```

It was generated with this CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.0)
set(CMAKE_TOOLCHAIN_FILE ./vcpkg/scripts/buildsystems/vcpkg.cmake)

project(calc)

add_executable(${PROJECT_NAME} calc.cpp)

find_package(unofficial-nana CONFIG REQUIRED)
target_link_libraries(${PROJECT_NAME} PRIVATE unofficial::nana::nana)
```

I'm by no means an experienced CMake user, but this CMakeLists.txt worked perfectly in Visual Studio on my Windows machine, so I don't think it's the problem. But then again, I wouldn't know.

--- a/ports/nana/CMakeLists.txt
+++ b/ports/nana/CMakeLists.txt
@@ -47,7 +47,7 @@ if(UNIX)
   target_include_directories(nana PUBLIC ${X11_INCLUDE_DIR})
   target_link_libraries(nana PUBLIC ${X11_LIBRARIES} ${X11_Xft_LIB})
 
-  find_package(Fontconfig)
+  find_package(Fontconfig REQUIRED)
   target_link_libraries(nana PUBLIC Fontconfig::Fontconfig)
 endif()
 

--- a/ports/nana/CMakeLists.txt
+++ b/ports/nana/CMakeLists.txt
@@ -48,7 +48,7 @@ if(UNIX)
   target_link_libraries(nana PUBLIC ${X11_LIBRARIES} ${X11_Xft_LIB})
 
   find_package(Fontconfig)
-  target_link_libraries(nana PUBLIC ${FONTCONFIG_LIB})
+  target_link_libraries(nana PUBLIC Fontconfig::Fontconfig)
 endif()
 
 if(NANA_ENABLE_PNG)

--- a/ports/nana/CMakeLists.txt
+++ b/ports/nana/CMakeLists.txt
@@ -47,7 +47,7 @@ if(UNIX)
   target_include_directories(nana PUBLIC ${X11_INCLUDE_DIR})
   target_link_libraries(nana PUBLIC ${X11_LIBRARIES} ${X11_Xft_LIB})
 
-  find_library(FONTCONFIG_LIB NAMES fontconfig)
+  find_package(Fontconfig)
   target_link_libraries(nana PUBLIC ${FONTCONFIG_LIB})
 endif()
 

--- a/ports/nana/CONTROL
+++ b/ports/nana/CONTROL
@@ -1,5 +1,6 @@
 Source: nana
 Version: 1.7.4
+Port-Version: 1
 Homepage: https://github.com/cnjinhao/nana
 Description: Cross-platform library for GUI programming in modern C++ style.
 Build-Depends: libpng, libjpeg-turbo, freetype (!uwp&&!windows), fontconfig (!uwp&&!windows)

--- a/ports/nana/config.cmake.in
+++ b/ports/nana/config.cmake.in
@@ -1,9 +1,10 @@
 @PACKAGE_INIT@
+include(CMakeFindDependencyMacro)
 
 if(UNIX)
-  find_package(Threads REQUIRED)
-  find_package(Freetype REQUIRED)
-  find_package(X11 REQUIRED)
+  find_dependency(Threads REQUIRED)
+  find_dependency(Freetype REQUIRED)
+  find_dependency(X11 REQUIRED)
   find_dependency(Fontconfig)
 endif()
 
@@ -11,11 +12,11 @@ option(NANA_ENABLE_PNG "Enable PNG support" @NANA_ENABLE_PNG@)
 option(NANA_ENABLE_JPEG "Enable JPEG support" @NANA_ENABLE_JPEG@)
 
 if(NANA_ENABLE_PNG)
-  find_package(PNG REQUIRED)
+  find_dependency(PNG REQUIRED)
 endif()
 
 if(NANA_ENABLE_JPEG)
-  find_package(JPEG REQUIRED)
+  find_dependency(JPEG REQUIRED)
 endif()
 
 include("${CMAKE_CURRENT_LIST_DIR}/unofficial-nana-targets.cmake")

--- a/ports/nana/config.cmake.in
+++ b/ports/nana/config.cmake.in
@@ -2,9 +2,9 @@
 include(CMakeFindDependencyMacro)
 
 if(UNIX)
-  find_dependency(Threads REQUIRED)
-  find_dependency(Freetype REQUIRED)
-  find_dependency(X11 REQUIRED)
+  find_dependency(Threads)
+  find_dependency(Freetype)
+  find_dependency(X11)
   find_dependency(Fontconfig)
 endif()
 
@@ -12,11 +12,11 @@ option(NANA_ENABLE_PNG "Enable PNG support" @NANA_ENABLE_PNG@)
 option(NANA_ENABLE_JPEG "Enable JPEG support" @NANA_ENABLE_JPEG@)
 
 if(NANA_ENABLE_PNG)
-  find_dependency(PNG REQUIRED)
+  find_dependency(PNG)
 endif()
 
 if(NANA_ENABLE_JPEG)
-  find_dependency(JPEG REQUIRED)
+  find_dependency(JPEG)
 endif()
 
 include("${CMAKE_CURRENT_LIST_DIR}/unofficial-nana-targets.cmake")

--- a/ports/nana/config.cmake.in
+++ b/ports/nana/config.cmake.in
@@ -4,7 +4,7 @@ if(UNIX)
   find_package(Threads REQUIRED)
   find_package(Freetype REQUIRED)
   find_package(X11 REQUIRED)
-  find_library(FONTCONFIG_LIB NAMES fontconfig)
+  find_dependency(Fontconfig)
 endif()
 
 option(NANA_ENABLE_PNG "Enable PNG support" @NANA_ENABLE_PNG@)
@@ -21,7 +21,7 @@ endif()
 include("${CMAKE_CURRENT_LIST_DIR}/unofficial-nana-targets.cmake")
 
 if(UNIX)
-  target_link_libraries(unofficial::nana::nana INTERFACE ${FONTCONFIG_LIB} ${X11_LIBRARIES} ${X11_Xft_LIB})
+  target_link_libraries(unofficial::nana::nana INTERFACE Fontconfig::Fontconfig ${X11_LIBRARIES} ${X11_Xft_LIB})
 endif()
 
 if(NANA_ENABLE_JPEG)

